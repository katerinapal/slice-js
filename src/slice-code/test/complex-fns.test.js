import {runAllCombosTests} from './helpers/utils'

// test.only(
//   'get("SomethingElse")',
//   require('./helpers/utils').snapSlice(require.resolve('./fixtures/complex-fns'), ({get}) => {
//     return [get('SomethingElse')]
//   })
// )

runAllCombosTests({
  filename: require.resolve('./fixtures/complex-fns'),
  methods: [
    {
      methodName: 'get',
      possibleArguments: [
        ['Luke'], ['Han'],
      ],
    },
  ],
})
