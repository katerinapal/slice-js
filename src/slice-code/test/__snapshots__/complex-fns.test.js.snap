exports[`test get("Han") 1`] = `
"export { unusedReturnAssign, get };

function unusedReturnAssign() {
  const foo = {};
  function bar(msg) {
    return function () {
      return \`bar: ${ msg }\`;
    };
  }
  foo.getLuke = bar(\'luke\');
  foo.getLuke.characterName = \'luke\';

  foo.getHan = bar(\'han\');
  foo.getHan.characterName = \'han\';

  return foo;
}

function get(name) {
  return unusedReturnAssign()[\`get${ name }\`]();
}"
`;

exports[`test get("Luke") && get("Han") 1`] = `
"export { unusedReturnAssign, get };

function unusedReturnAssign() {
  const foo = {};
  function bar(msg) {
    return function () {
      return \`bar: ${ msg }\`;
    };
  }
  foo.getLuke = bar(\'luke\');
  foo.getLuke.characterName = \'luke\';

  foo.getHan = bar(\'han\');
  foo.getHan.characterName = \'han\';

  return foo;
}

function get(name) {
  return unusedReturnAssign()[\`get${ name }\`]();
}"
`;

exports[`test get("Luke") 1`] = `
"export { unusedReturnAssign, get };

function unusedReturnAssign() {
  const foo = {};
  function bar(msg) {
    return function () {
      return \`bar: ${ msg }\`;
    };
  }
  foo.getLuke = bar(\'luke\');
  foo.getLuke.characterName = \'luke\';

  foo.getHan = bar(\'han\');
  foo.getHan.characterName = \'han\';

  return foo;
}

function get(name) {
  return unusedReturnAssign()[\`get${ name }\`]();
}"
`;
